src/components/forms/SelectField/AsyncSelectField.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import {useAsync} from 'react-use';
+
+import {SelectField} from './SelectField';
+
+const withOptions = WrappedComponent => {
+  return ({optionsRetriever, ...props}) => {
+    const {loading, error, value: options} = useAsync(optionsRetriever);
+
+    if (loading) {
+      return (
+        <WrappedComponent
+          options={[{label: 'Loading...', value: null}]} // Display loading message
+          {...props}
+        />
+      );
+    }
+
+    if (error) {
+      return <div>Error loading options: {error.message}</div>; // Display error message
+    }
```

Comment:
"nice" in a way that it doesn't just crash, but ultimately unhelpful for the end-user because they have no way of correcting the problem.

So with that in mind, I'd favour just throwing the exception like [here](https://github.com/open-formulieren/formio-builder/blob/main/src/components/builder/prefill/plugin.tsx#L36) and let the error boundary deal with it in a generic way.

In the future, we may support an `onError` prop so that the calling component can define retry behaviour or what to do when loading fails. (so don't add this now).